### PR TITLE
chore: refactor hardhat tests

### PR DESCRIPTION
**Pull request by zoruka**
- Rearrange hardhat tests folder structure;
- Split out the single FleekERC721 suit to multiple files.

### CODE REVIEW

File: contracts/test/hardhat/contracts/FleekERC721/deployment.t.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { loadFixture } from '@nomicfoundation/hardhat-network-helpers';
+import { expect } from 'chai';
+import { TestConstants, Fixtures } from './helpers';
+
+describe('FleekERC721.Deployment', () => {
+  it('should assign the name and the symbol of the ERC721 contract', async () => {
+    const { contract } = await loadFixture(Fixtures.default);
```
Comment (jsonsivar) [inline contracts/test/hardhat/contracts/FleekERC721/deployment.t.ts:7]: Can this go into a beforeEach?

File: contracts/test/hardhat/contracts/FleekERC721/access-points.t.ts [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+      .to.emit(contract, 'NewAccessPoint')
+      .withArgs('accesspoint.com', tokenId, owner.address);
+
+    expect(await contract.appAccessPoints(tokenId)).eql(['accesspoint.com']);
+  });
+
+  it('should return a AP json object', async () => {
+    const { contract, owner, tokenId } = fixture;
+
+    await contract.addAccessPoint(tokenId, 'accesspoint.com');
```
Comment (jsonsivar) [inline contracts/test/hardhat/contracts/FleekERC721/access-points.t.ts:25]: Should this go into the beforeEach?

File: contracts/test/hardhat/contracts/FleekERC721/token-uri.t.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
...gzMi41NTgiIGZpbHRlclVuaXRzPSJ1c2VyU3BhY2VPblVzZSIgY29sb3ItaW50ZXJwb2xhdGlvbi1maWx0ZXJzPSJzUkdCIj48ZmVGbG9vZCBmbG9vZC1vcGFjaXR5PSIwIiAvPjxmZUJsZW5kIGluPSJTb3VyY2VHcmFwaGljIiAvPjxmZUdhdXNzaWFuQmx1ciBzdGREZXZpYXRpb249IjQyIiAvPjwvZmlsdGVyPjxsaW5lYXJHcmFkaWVudCBpZD0iYmFja2dyb3VuZCIgeDE9IjUzMi41IiB5MT0iMCIgeDI9IjUzMi41IiB5Mj0iMTA2NSIgZ3JhZGllbnRVbml0cz0idXNlclNwYWNlT25Vc2UiPjxzdG9wIC8+PHN0b3Agb2Zmc2V0PSIxIiBzdG9wLWNvbG9yPSIjMTMxMzEzIiAvPjwvbGluZWFyR3JhZGllbnQ+PHJhZGlhbEdyYWRpZW50IGlkPSJiYWNrZ3JvdW5kLXJhZGlhbCIgY3g9IjAiIGN5PSIwIiByPSIxIiBncmFkaWVudFVuaXRzPSJ1c2VyU3BhY2VPblVzZSIgZ3JhZGllbnRUcmFuc2Zvcm09InRyYW5zbGF0ZSg1MzIuNSA1MzIuNSkgcm90YXRlKDg5Ljk2MSkgc2NhbGUoNzM1KSI+PHN0b3Agc3RvcC1jb2xvcj0iI2UzNGYyNiIgLz48c3RvcCBvZmZzZXQ9IjEiIHN0b3AtY29sb3I9IiNlMzRmMjYiIHN0b3Atb3BhY2l0eT0iMCIgLz48L3JhZGlhbEdyYWRpZW50PjxsaW5lYXJHcmFkaWVudCBpZD0iZGlza2V0dGUtZ3JhZGllbnQiIHgxPSI5MjUuNjI2IiB5MT0iMjU2Ljg5NiIgeDI9IjEzNi43NzkiIHkyPSI4MDAuMjAzIiBncmFkaWVudFVuaXRzPSJ1c2VyU3BhY2VPblVzZSI+PHN0b3Agc3RvcC1jb2xvcj0iI2UzNGYyNiIgLz48c3RvcCBvZmZzZXQ9IjEiIHN0b3AtY29sb3I9IiMyQzMxM0YiIC8+PC9saW5lYXJHcmFkaWVudD48bGluZWFyR3JhZGllbnQgaWQ9Im1haW4iPjxzdG9wIHN0b3AtY29sb3I9IiNlMzRmMjYiIC8+PC9saW5lYXJHcmFkaWVudD48L2RlZnM+PC9zdmc+',
```
Comment (jsonsivar) [inline contracts/test/hardhat/contracts/FleekERC721/token-uri.t.ts:23]: Maybe this can come from another file or constants?

**Review by jsonsivar — DISMISSED** [review]
LGTM but added some minor comments.

**Review by jsonsivar — APPROVED** [review]
LGTM.  Let's merge this? @EmperorOrokuSaki are you still planning to review it?

**Comment by EmperorOrokuSaki** [comment]
> LGTM. Let's merge this? @EmperorOrokuSaki are you still planning to review it?

Yes I had reviewed this while working on hardhat tests for the auto approval PR.

**Review by EmperorOrokuSaki — APPROVED** [review]

**Pull request merged**